qm-plugins/qm-cron/class-qm-cron-output-html.php

```diff
@@ -0,0 +1,333 @@
+
+		echo '<td' . $row_span . '>', esc_html( $hook ), '</td>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+
+	/**
+	* Displays the the event schedule name for recurring events or else 'single event'.
+	*
+	* @param array $info Event info array.
+	*/
+	private function display_event_schedule( $info ) {
+		if ( ! empty( $info['schedule'] ) ) {
+			echo esc_html( $info['schedule'] );
+		} else {
+			esc_html_e( 'Single Event', 'query-monitor' );
+		}
+	}
+
+
+	/**
+	* Displays the event interval in seconds, minutes and hours.
+	*
+	* @param array $info Event info array.
+	*/
+	private function display_event_intervals( $info ) {
+		if ( ! empty( $info['interval'] ) ) {
+			$interval = (int) $info['interval'];
+			/* translators: %s is number of seconds. */
+			printf( esc_html__( '%ss', 'query-monitor' ) . '<br />', esc_html( $interval ) );
+			/* translators: %s is number of minutes. */
+			printf( esc_html__( '%sm', 'query-monitor' ) . '<br />', esc_html( $this->get_minutes( $interval ) ) );
+			/* translators: %s is number of hours. */
+			printf( esc_html__( '%sh', 'query-monitor' ), esc_html( $this->get_hours( $interval ) ) );
+			unset( $interval );
+		} else {
+			esc_html_e( 'Single Event', 'query-monitor' );
+		}
+	}
+
+	/**
+	* Displays the cron arguments in a readable format.
+	*
+	* @param mixed $args Cron argument(s).
+	*
+	* @return void
+	*/
+	private function display_event_cron_arguments( $args ) {
+		// Arguments defaults to an empty array if no arguments are given.
+		if ( is_array( $args ) && array() === $args ) {
+			esc_html_e( 'No Args', 'query-monitor' );
+			return;
+		}
+
+		echo wp_json_encode( $args );
+	}
+
+	/**
+	 * Verify if a given timestamp is in the past or the future.
+	 *
+	 * @param int $time Unix timestamp.
+	 *
+	 * @return bool True if the time has passed, false otherwise.
+	 */
+	private function is_time_in_past( $time ) {
+		return ( time() > $time && false === $this->doing_cron );
+	}
+
+	/**
+	 * Compares time with current time and adds ' ago' if current time is greater than event time.
+	 *
+	 * @param string $human_time Human readable time difference.
+	 * @param int    $time       Unix time of event.
+	 *
+	 * @return string
+	 */
+	private function display_past_time( $human_time, $time ) {
+		if ( time() > $time ) {
+			/* translators: %s is a human readable time difference. */
+			return sprintf( esc_html__( '%s ago', 'query-monitor' ), $human_time );
+		} else {
+			return $human_time;
+		}
+	}
+
+	/**
+	 * Transform a time in seconds to minutes rounded to 2 decimals.
+	 *
+	 * @param int $time Unix timestamp.
+	 *
+	 * @return float
+	 */
+	private function get_minutes( $time ) {
+		return round( ( (int) $time / 60 ), 2 );
+	}
+
+
+	/**
+	 * Transform a time in seconds to hours rounded to 2 decimals.
+	 *
+	 * @param int $time Unix timestamp.
+	 *
+	 * @return float
+	 */
+	private function get_hours( $time ) {
+		return round( ( (int) $time / 3600 ), 2 );
+	}
```

Comment:
I think I have already seen exactly the same functions in `class-qm-cron-collector.php`.

Do you think it makes sense to create a helper class and move these methods into it to avoid code duplication?

Reply:
Ah, whoops. I actually intended to move it and accidentally left the one in the other file. I've removed the duplicated one.